src/librawspeed/decoders/NakedDecoder.cpp

```diff
@@ -44,50 +44,55 @@ NakedDecoder::NakedDecoder(FileMap* file, Camera* c) :
 
 NakedDecoder::~NakedDecoder() = default;
 
-RawImage NakedDecoder::decodeRawInternal() {
-  uint32 width=0, height=0, filesize=0, bits=0, offset=0;
-  if(cam->hints.find("full_width") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("full_width"))->second;
-    width = (uint32) atoi(tmp.c_str());
-  } else
-    ThrowRDE("Naked: couldn't find width");
-
-  if(cam->hints.find("full_height") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("full_height"))->second;
-    height = (uint32) atoi(tmp.c_str());
-  } else
-    ThrowRDE("Naked: couldn't find height");
-
-  if(cam->hints.find("filesize") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("filesize"))->second;
-    filesize = (uint32) atoi(tmp.c_str());
-  } else
-    ThrowRDE("Naked: couldn't find filesize");
-
-  if(cam->hints.find("offset") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("offset"))->second;
-    offset = (uint32) atoi(tmp.c_str());
-  }
+static const map<string, BitOrder> order2enum = {
+    {"plain", BitOrder_Plain},
+    {"jpeg", BitOrder_Jpeg},
+    {"jpeg16", BitOrder_Jpeg16},
+    {"jpeg32", BitOrder_Jpeg32},
+};
+
+void NakedDecoder::parseHints() {
```

Comment:
I'm sorry, but we will have to disagree here. I do not think the code that was deleted by these changes was better than the new one.
Separation into smaller, shorter functions, and into smaller translation units is more readable.
It is easier to understand what each piece does, if the pieces are mixed in a one messy function / large .cpp
It shouldl eventually be easier to test such code, too